src/app/pages/sharing/components/shares-dashboard/shares-dashboard.component.ts

```diff
@@ -167,10 +163,29 @@ export class SharesDashboardComponent implements AfterViewInit {
           ],
           detailsHref: '/sharing/nfs',
           add() {
-            this.parent.add(this.tableComponent, ShareType.Nfs);
+            this.parent.slideInService.open(NfsFormComponent);
+            this.parent.slideInService.onClose$
```

Comment:
The code inside the `onClose$` callback looks similar but it isn't really. `this` inside the `add` method doesn't point to `SharesDashboardComponent`. It points to the `TableComponent` inside the `ExpandableTableComponent` that we have 4 of in the `SharesDashboardComponent`. Meaning in each of the `add` methods, the access we have to the `this.getData` method, we won't have outside the `add` method. We can subscribe to it once in someplace else but the `SlideInService` doesn't really has nice way of communicating what page it's closing. It's configured to close with a response or error object. But that response is meant to be a response from the form submission. Not the form identifying itself. It can be done but it's unclean. Which is why I was avoiding that particular route.

Reply:
I've extended the `IxSlideinService` to return the modal type when it closes so we have a way of knowing what form just closed.